corehq/pillows/reportcase.py

```diff
@@ -28,13 +30,20 @@ def get_unique_id(cls):
         return '8c10a7564b6af5052f8b86693bf6ac07'
 
     def change_transform(self, doc_dict):
-        return transform_case_to_report_es(doc_dict)
+        if not report_case_filter(doc_dict):
+            return transform_case_to_report_es(doc_dict)
 
 
-def transform_case_to_report_es(doc_dict):
+def report_case_filter(doc_dict):
+    """
+        :return: True to filter out doc
+    """
     if doc_dict.get('domain', None) not in getattr(settings, 'ES_CASE_FULL_INDEX_DOMAINS', []):
         # full indexing is only enabled for select domains on an opt-in basis
-        return None
+        return True
```

Comment:
Nit: could simplify to `return doc_dict.get('domain') not in getattr(settings, 'ES_CASE_FULL_INDEX_DOMAINS', [])`